composition-js/src/compose.ts

```diff
@@ -45,66 +45,105 @@ function validateCompositionOptions(options: CompositionOptions) {
+ * 
+ * @param subgraphs Subgraphs
+ * @param options CompositionOptions
+ * @param runSatisfiability Boolean - flag used to toggle satisfiability. Defaults to `true`
+ */
+export function compose(subgraphs: Subgraphs, options: CompositionOptions = {}, runSatisfiability = true): CompositionResult {
   validateCompositionOptions(options);
 
-  const upgradeResult = upgradeSubgraphsIfNecessary(subgraphs);
-  if (upgradeResult.errors) {
-    return { errors: upgradeResult.errors };
-  }
-
-  const toMerge = upgradeResult.subgraphs;
-  const validationErrors = toMerge.validate();
-  if (validationErrors) {
-    return { errors: validationErrors };
-  }
-
-  const mergeResult = mergeSubgraphs(toMerge);
+  const mergeResult = validateSubgraphsAndMerge(subgraphs);
   if (mergeResult.errors) {
     return { errors: mergeResult.errors };
   }
 
-  // We pass `null` for the `supportedFeatures` to disable the feature support validation. Validating feature support
-  // is useful when executing/handling a supergraph, but here we're just validating the supergraph we've just created,
-  // and there is no reason to error due to an unsupported feature.
-  const supergraph = new Supergraph(mergeResult.supergraph, null);
-  const supergraphQueryGraph = buildSupergraphAPIQueryGraph(supergraph);
-  const federatedQueryGraph = buildFederatedQueryGraph(supergraph, false);
-  const { errors, hints } = validateGraphComposition(
-    supergraph.schema,
-    supergraph.subgraphNameToGraphEnumValue(),
-    supergraphQueryGraph,
-    federatedQueryGraph
-  );
-  if (errors) {
-    return { errors };
+  let satisfiabilityResult;
+  if (runSatisfiability) {
+    satisfiabilityResult = validateSatisfiability({
+      supergraphSchema: mergeResult.supergraph
+    });
+    if (satisfiabilityResult.errors) {
+      return { errors: satisfiabilityResult.errors };
+    }
   }
 
   // printSchema calls validateOptions, which can throw
   let supergraphSdl;
   try {
     supergraphSdl = printSchema(
-      supergraph.schema,
+      mergeResult.supergraph,
       options.sdlPrintOptions ?? shallowOrderPrintedDefinitions(defaultPrintOptions),
     );
   } catch (err) {
     return { errors: [err] };
   }
 
   return {
-    schema: supergraph.schema,
+    schema: mergeResult.supergraph,
     supergraphSdl,
-    hints: mergeResult.hints.concat(hints ?? []),
+    hints: [...mergeResult.hints, ...(satisfiabilityResult?.hints ?? [])],
   };
 }
 
-export function composeServices(services: ServiceDefinition[], options: CompositionOptions = {}): CompositionResult  {
+/**
+ * Method to validate and compose services
+ * 
+ * @param services List of Service definitions
+ * @param options CompositionOptions
+ * @param runSatisfiability Flag to toggle satisfiability check within composition. Defaults to `true`
+ * @returns CompositionResult
+ */
+export function composeServices(services: ServiceDefinition[], options: CompositionOptions = {}, runSatisfiability = true): CompositionResult  {
```

Comment:
Why don't we just add this to `CompositionOptions` rather than having an extra argument?

Reply:
addressed on 04db517c